tests/providers/azure/internal/test_common.py

```diff
@@ -28,6 +28,13 @@ class AZCommonTest(unittest.TestCase):
   """Test Azure common file."""
   # pylint: disable=line-too-long
 
+  # Stop tests from picking up Azure creds from host.
+  @typing.no_type_check
+  def setUp(self):
+    os.environ['AZURE_CONFIG_DIR'] = os.path.join(
+        os.path.dirname(os.path.dirname(os.path.dirname(os.path.dirname(
+            os.path.realpath(__file__))))), azure_mocks.EMPTY_AZURE_CONFIG_DIR)
```

Comment:
I would suggest assigning the filename to its own variable to make the `join` more readable